OsmAnd-java/src/main/java/net/osmand/search/SearchUICore.java

```diff
@@ -304,7 +304,7 @@ public boolean sameSearchResult(SearchResult r1, SearchResult r2) {
 					return MapUtils.getDistance(r1.location, r2.location) < similarityRadius;
 				}
 			} else if (r1.object != null && r2.object != null) {
-				return r1.object == r2.object;
+				return r1.object.equals(r2.object);
```

Comment:
it's comparision for same object not for similar object, so this is incorrect and we don't support equals on all types